JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -211,41 +204,57 @@ private static void GetAndSetTokens(Map<String, String> formData) throws Excepti
 	    	    String res = buf.toString("UTF-8");
 	        	JsonElement jElement = new JsonParser().parse(res);
 	            JsonObject jObject = jElement.getAsJsonObject();
-	            
 				// set Tokens as System Properties - new values to access_token and refresh_token
-				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, jObject.get("access_token").getAsString());
-				System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, jObject.get("refresh_token").getAsString());
-
+	            if(null == jObject.get("access_token").getAsString().isEmpty() || jObject.get("access_token").getAsString().isEmpty())
+	            {
+	            	throw new Exception("Access Token is empty, Please verify the config.properties for proper inputs");
+	            }else 
+	            {
+					System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, jObject.get("access_token").getAsString());
+	            }
+	            if(null == jObject.get("refresh_token").getAsString().isEmpty() || jObject.get("refresh_token").getAsString().isEmpty())
+				{	
+					throw new Exception("Refresh Token is empty, Please verify the config.properties for proper inputs");
+	            }else
+				{
+					System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, jObject.get("refresh_token").getAsString());
+				}
 	        } catch (Exception e) {
 	            String res = "";
-	            try {
+	            try 
+	            {
 	                BufferedInputStream in = new BufferedInputStream(connection.getErrorStream());
 	                ByteArrayOutputStream buf = new ByteArrayOutputStream();
-	                for (int result = in.read(); result != -1; result = in.read()) {
+	                for (int result = in.read(); result != -1; result = in.read()) 
```

Comment:
This code section, reading from the input stream, should be made in its own function since it is an exact repeat of the code around line 200.

Reply:
Added the function to perform copy task.